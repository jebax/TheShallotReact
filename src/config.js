export const config = {
  guardianUrl: `http://content.guardianapis.com/search?section=technology&show-fields=thumbnail&order-by=newest&page-size=10&api-key=${process.env.REACT_APP_GUARDIAN_KEY}`,
  aylienUrl: 'https://cors-anywhere.herokuapp.com/https://api.aylien.com/api/v1/summarize?url='
}
