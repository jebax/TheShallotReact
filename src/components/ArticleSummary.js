import React, { Component } from 'react'
import axios from 'axios'
import { config } from '../config'

export default class ArticleSummary extends Component {
  state = {
    sentences: []
  }

  componentDidMount() {
    axios.get(`${config.aylienUrl}${this.props.location.state.url}`, {
      headers: {
        'X-AYLIEN-TextAPI-Application-Key': process.env.REACT_APP_AYLIEN_KEY,
        'X-AYLIEN-TextAPI-Application-ID': process.env.REACT_APP_AYLIEN_ID
      }
    })
    .then(response => {
      this.setState(prevState => ({sentences: response.data.sentences}))
    })
  }

  render() {
    return (
      <a
        className='summary-headline'
        href={this.props.location.state.url}
      >
        {this.props.location.state.headline}
      </a>

    )
  }
}
