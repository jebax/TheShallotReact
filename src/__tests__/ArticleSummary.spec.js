import React from 'react'
import { shallow } from 'enzyme'
import axios from 'axios'
import ArticleSummary from '../components/ArticleSummary'
import { config } from '../config'

jest.mock('axios')

describe('<ArticleSummary />', () => {
  var wrapper

  beforeAll(() => {
    axios.get.mockResolvedValue({})
    const location = {
      state: {
        id: 0,
        headline: 'TestHeadline',
        url: 'TestUrl'
      }
    }
    wrapper = shallow(<ArticleSummary location={location} />)
  })

  it('matches the snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('has a link to an article', () => {
    expect(wrapper.find('a').length).toEqual(1)
  })

  it('has a link based on its state', () => {
    const link = wrapper.find('a')

    expect(link.text()).toEqual('TestHeadline')
    expect(link.props().href).toEqual('TestUrl')
  })

  it('makes a request to the correct API with the related URL', () => {
    expect(axios.get).toHaveBeenCalledTimes(1)
    expect(axios.get).toHaveBeenCalledWith(`${config.aylienUrl}TestUrl`, {
      headers: {}
    })
  })
})
