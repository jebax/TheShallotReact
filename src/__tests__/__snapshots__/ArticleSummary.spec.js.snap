// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ArticleSummary /> matches the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ArticleSummary
    location={
      Object {
        "state": Object {
          "headline": "TestHeadline",
          "id": 0,
          "url": "TestUrl",
        },
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": "TestHeadline",
      "className": "summary-headline",
      "href": "TestUrl",
    },
    "ref": null,
    "rendered": "TestHeadline",
    "type": "a",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "TestHeadline",
        "className": "summary-headline",
        "href": "TestUrl",
      },
      "ref": null,
      "rendered": "TestHeadline",
      "type": "a",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
